### PR TITLE
nanomsgxx 0.1 (new formula)

### PR DESCRIPTION
**Pull request by piribes**
### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?

_You can erase any parts of this template not applicable to your Pull Request._

### New Formulae Submissions:

- [x] Does your submission pass
`brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Have you built your formula locally prior to submission with `brew install <formula>`?

### Changes to Homebrew's Core:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031) if you'd like one.
- [x] Have you successfully ran `brew tests` with your changes locally?

### Description of nanomsgxx

nanomsgxx aims to provide a very thin abstraction on top of the nanomsg C API, while taking advantage of C++11's features to make the code easier to read and write.

Despite the version number, [according to the author](https://github.com/achille-roussel/nanomsgxx/releases), this is a stable release.
> This is the first release of nanomsgxx, it's mostly stable as it's known to be used by a bunch of people. While this is not a 1.x version it's really unlikely that breaking changes will be brought to the API.

### Known Audit Problems

 * GitHub repository not notable enough (<20 forks, <20 watchers and <50 stars)

### CODE REVIEW

**Comment by tdsmith** [comment]
Only CI failure is the audit.

The codebase looks a few years old, so "first release" may not completely describe its maturity.

File: Library/Formula/nanomsgxx.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+class Nanomsgxx < Formula
+  desc "Nanomsg binding for C++11"
+  homepage "https://achille-roussel.github.io/nanomsgxx/doc/nanomsgxx.7.html"
+  url "https://github.com/achille-roussel/nanomsgxx/archive/0.1.tar.gz"
+  sha256 "50f6d2507f264f048e0234e78bbbfe80004d636d990fefc4e1f53a5434c4aa63"
+
+  option "with-debug", "Compile with debug symbols"
+
+  depends_on "pkg-config" => :build
+  depends_on :python
```
Comment (DomT4) [inline Library/Formula/nanomsgxx.rb]: Should be `depends_on :python if MacOS.version <= :snow_leopard` I suspect. Actually needed for run or only build?

File: Library/Formula/nanomsgxx.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    args << "--debug" if build.with? "debug"
+
+    system "python", "./waf", "configure", *args
+    system "python", "./waf", "build"
+    system "python", "./waf", "install"
+  end
+
+  test do
+    (testpath/"test.cpp").write <<-EOS
```
Comment (DomT4) [inline Library/Formula/nanomsgxx.rb]: Please use `<<-EOS.undent` and then indent the below line. Looks a fair bit neater.

File: Library/Formula/nanomsgxx.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+#include <iostream>\nint main(int argc, char **argv) {\n\tstd::cout << "Hello Nanomsgxx!" << std::endl;\n}
+    EOS
+
+    system "c++", "-std=c++11", "-lnnxx", "test.cpp"
```
Comment (DomT4) [inline Library/Formula/nanomsgxx.rb]: Use `ENV.cxx` rather than `system "c++"`

File: Library/Formula/nanomsgxx.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    assert_equal "Hello Nanomsgxx!\n",
+        shell_output("#{testpath}/a.out")
```
Comment (DomT4) [inline Library/Formula/nanomsgxx.rb]: This is short enough to be on one line rather than two.

File: Library/Formula/nanomsgxx.rb (hunk 1 above)
Comment (piribes) [inline Library/Formula/nanomsgxx.rb]: Only for build

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
Popularity is the only failure at this point.

**Comment by apjanke** [comment]
@BrewTestBot test this please

**Comment by apjanke** [comment]
Audit is still failing due to notability.

```
failed: brew audit nanomsgxx --strict --online

Stacktrace
        Error: 1 problem in 1 formula
nanomsgxx:
  * GitHub repository not notable enough (<20 forks, <20 watchers and <50 stars)
```

At this point, we can't accept this in core Homebrew. But if the project becomes more popular, feel free to re-submit this on the new [`homebrew/core` repo](https://github.com/homebrew/homebrew-core). In the meantime, you can create your own tap for it.

**Pull request closed without merging**